mne/utils/fetching.py

```diff
@@ -22,18 +22,9 @@ def _get_http(url, temp_file_name, initial_size, file_size, timeout,
     # Actually do the reading
     req = request.Request(url)
     if initial_size > 0:
+        logger.debug('  Resuming at %s' % (initial_size,))
         req.headers['Range'] = 'bytes=%s-' % (initial_size,)
-    try:
-        response = request.urlopen(req, timeout=timeout)
-    except Exception:
-        # There is a problem that may be due to resuming, some
-        # servers may not support the "Range" header. Switch
-        # back to complete download method
-        logger.info('Resuming download failed (server '
-                    'rejected the request). Attempting to '
-                    'restart downloading the entire file.')
-        del req.headers['Range']
-        response = request.urlopen(req, timeout=timeout)
```

Comment:
this was done for a reason in nilearn originally. Yes copied code from there. Is it still like this in nilearn?

Reply:
The reason to do it is presumably because some servers don't support resuming ([and yes they still have a variant of it](https://github.com/nilearn/nilearn/blob/35f4906ca37a5d46378ad1c9fe530bd0cea442ff/nilearn/datasets/utils.py#L545-L552)). The problem is that this code can fail for non-resuming related reasons (such as timeout, or URL not found), in which case we just blindly try again. So if we know the servers we use support it, then we're better off not doing this.

Reply:
The function in nilearn has changed qui te a bit. https://github.com/nilearn/nilearn/blob/759102386928a7c869b1299ede3e9ce28896ec35/nilearn/datasets/utils.py#L450

This try catch was introduced [in 2012 here](https://github.com/nilearn/nilearn/commit/651a7063d8f3b477dbe8d07d551b150ef9e9ec31) and removed [in 2015 here](https://github.com/nilearn/nilearn/commit/c90df18f08e5559c60ceac3366b8b13443a54e60) I don't think i makes anylonger sense.